File: py-appscript/branches/py-appscript-0.16.2/Examples/appscript/Add_iCal_event.py
#!/usr/bin/env python

import datetime
from appscript import *

# Add event to Home calendar to run from 7pm to 9 pm today
# (note: when only time is given, appscript uses current date)

calendarName = 'Home'
start = datetime.time(19, 0, 0) # 7 pm
end = datetime.time(21, 0, 0) # 9 pm
summary = 'First pants, then shoes.'

app('iCal').calendars[its.title == calendarName].events.end.make(
        new=k.event, with_properties={
                k.start_date: start, 
                k.end_date: end, 
                k.summary: summary})

File: rb-appscript/tags/rb-appscript-0.5.2/sample/Add_iCal_event.rb
#!/usr/bin/env ruby

# Add an event to Home calendar that runs from 7am to 9 am tomorrow

# Note: if using the appscript gem, rubygems must be required first:
begin; require 'rubygems'; rescue LoadError; end

require 'appscript'
include Appscript

calendar_name = 'Home'
t = Time.now + 60 * 60 * 24
start = Time.local(t.year, t.month, t.day, 7)
end_ = start + 60 * 60 * 2
summary = 'First pants, then shoes.'

app('iCal').calendars[calendar_name].events.end.make(
        :new => :event, :with_properties => {
                :start_date => start, 
                :end_date => end_, 
                :summary => summary})
